### PR TITLE
Add Alerts utilities for local and global feedback

### PR DESCRIPTION
**Pull request by davidebriani**
- Add useAlerts() hook to produce and display in-page contextual feedback
- Add GlobalAlertsProvider + useGlobalAlerts() hook to produce app-wide feedback and display it in a centralized place

Signed-off-by: Davide Briani <davide.briani@ispirata.com>

### CODE REVIEW

File: src/react/AlertManager.js [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+  const [alerts, setAlerts] = useState([]);
+
+  useEffect(() => {
+    const timersIdRef = timersId.current;
+    return () => timersIdRef.forEach(clearTimeout);
+  }, []);
+
+  const close = useCallback(
+    (alert) => {
+      setAlerts((currentAlerts) => {
```
Comment (Pavinati) [inline src/react/AlertManager.js:43]: why checking for different array length when you discard the alert anyway?
```
setAlerts((currentAlerts) => {
  if (alert.options.onClose) {
    alert.options.onClose();
  }
  return currentAlerts.filter((a) => a.id !== alert.id);
}
```

File: src/react/AlertManager.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        if (
+          lengthBeforeRemove > filteredAlerts.length &&
+          alert.options.onClose
+        ) {
+          alert.options.onClose();
+        }
+        return filteredAlerts;
+      });
+    },
+    [setAlerts]
```
Comment (Pavinati) [inline src/react/AlertManager.js:55]: `setAlerts` is guaranteed to have a constant ref, no need to add it to the dependency list

File: src/react/AlertManager.js [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+    showSuccess,
+    showWarning,
+    showError,
+    showInfo,
+  };
+
+  return (
+    <GlobalAlertsContext.Provider value={alertContext} {...props}>
+      {children}
+      {root.current && createPortal(<Alerts />, root.current)}
```
Comment (Pavinati) [inline src/react/AlertManager.js:220]: Here you have a context provider that also injects elements in the DOM. If on a later that we will move alerts on a different level of the layout, this code will need some heavy refactoring.

File: src/react/AlertManager.js (hunk 1 above)
Comment (davidebriani) [reply src/react/AlertManager.js:43]: This hook exposes the alerts array, so you could get a reference on each alert and call `alert.close()` multiple times (by mistake) but you would expect it to be closed just once. `alert.onClose()` should be called only once as well

File: src/react/AlertManager.js (hunk 2 above)
Comment (davidebriani) [reply src/react/AlertManager.js:55]: That's a good point

File: src/react/AlertManager.js (hunk 3 above)
Comment (davidebriani) [reply src/react/AlertManager.js:220]: Good point. I'll refactor it so we can place the Alerts where we need them.

**Review by Pavinati — APPROVED** [review]

**Pull request merged**